sisyphus/ci.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sisyphus
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	u "istio.io/test-infra/toolbox/util"
+)
+
+// CI defines common accessors for continuous integration systems
+type CI interface {
+	GetLatestRun(jobName string) (int, error)
+	GetResult(jobName string, runNo int) (*Result, error)
+	Rerun(jobName string, runNo, numRerun int) error
+	GetDetailsURL(jobName string, runNo int) string
+}
+
+// Result defines the output of each CI run
+type Result struct {
+	Passed bool   `json:"passed"`
+	SHA    string `json:"sha"`
+}
+
+const (
+	lastBuildTXT = "latest-build.txt"
+	finishedJSON = "finished.json"
+	startedJSON  = "started.json"
+)
+
+// ProwResult matches the structure published in finished.json
+type ProwResult struct {
+	TimeStamp  uint32       `json:"timestamp"`
+	Version    string       `json:"version"`
+	Result     string       `json:"result"`
+	Passed     bool         `json:"passed"`
+	JobVersion string       `json:"job-version"`
+	Metadata   ProwMetadata `json:"metadata"`
+}
+
+// ProwMetadata matches the structure published in finished.json
+type ProwMetadata struct {
+	Repo       string `json:"repo"`
+	RepoCommit string `json:"repo-commit"`
+}
+
+// ProwJobConfig matches the structure published in started.json
+type ProwJobConfig struct {
+	Node        string `json:"node"`
+	JenkinsNode string `json:"jenkins-node"`
+	Version     string `json:"version"`
```

Comment:
Are you using the JenkinsNode ? if not you can remove it, json parsing will do the right thing.